### PR TITLE
MNT: Build from VCS source and remove cython upper bound

### LINKED ISSUE #21: Is cython upper bound required?
### Comment:

In PR https://github.com/conda-forge/lhapdf-feedstock/pull/18 an upper bound on `cython` `v3` was added for both the `build` and `host` requirements

https://github.com/conda-forge/lhapdf-feedstock/blob/5cd40801ba1be899220dcb85d3259ed779ae7bd6/recipe/meta.yaml#L20-L23

https://github.com/conda-forge/lhapdf-feedstock/blob/5cd40801ba1be899220dcb85d3259ed779ae7bd6/recipe/meta.yaml#L30-L32

It wasn't super clear from the PR why this was necessary. Is this still required?

Upper bounds on library dependencies can lead to breakage or installation problems in the future, so more curious if this can get changed to a _lower bound_ to avoid breakage, or if there is a fundamental incompatibility with the `lhapdf` source code with `cython` `v3` that would require a change in the project code itself.

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Resolves #21
* Closes #22 

## Description

* Use GitLab as the source for a consistency between development and build source as hepforge releases have configure scripts pre-buit which need to be rebuilt in certain cases, so it is preferable to build everything consistently.
   - c.f. https://gitlab.com/hepcedar/lhapdf
* Add `automake`, `autoconf`, and `libtool` as `build` requirements. These are all necessary for working with `autotools` (e.g. `autoconf` and `autoreconf`).
* Use `autoreconf --install --force` for all builds to generate the `configure` script.
* Apply the `osx_patch.diff` patch for macOS selectively by defining it as a patch in `recipe/meta.yaml`.
* Add `cython >=0.24` as both a `host` requirement and as a `build` requirement for cross compiles. This removes the upper bound allowing for Cython `v3.x` to be used, but only enforces the lower bound of Cython `v0.24` that `configure` checks for.
* Use `cython` to build the sources for the Python bindings for all builds as this can be useful in cases like new Python versions
* Use `cython`'s `--include-dir` option to properly pick up the include files for the C++ file generation for the Python bindings.

  ```
  -I INCLUDE_PATH, --include-dir INCLUDE_PATH
                        Search for include files in named directory (multiple
                        include directories are allowed).
  ```

* Add `CXXFLAGS` to configure which affects `lhapdf-config --cxxflags`.
* Add reference to install docs on why `-Qunused-arguments` are used for macOS builds.
* Bump build number.

* Add test environment requirements and files from the source tar.gz.
   - c.f. https://conda-forge.org/docs/maintainer/adding_pkgs/#test
* Add `cxx-compiler` and `numpy` as 'test' requirements.
* Add post build tests that run recipe/run_test.sh.
   - Tests the installed directory structure, that necessary library files exist, the lhapdf-config and lhapdf CLI API, and builds and runs C++ and Python LHAPDF examples.

c.f https://conda-forge.org/docs/maintainer/knowledge_base/#cross-compilation-examples

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [N/A] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh
Comment (matthewfeickert) [inline recipe/build.sh]: @chrisburr This works, which is great, but can you explain why this is needed?

File: recipe/run_test.sh
Comment (matthewfeickert) [inline recipe/run_test.sh]: Currently failing on macOS with

```
# Test example analyticpdf
dyld[17941]: Library not loaded: '@rpath/libLHAPDF.dylib'
  Referenced from: '/Users/runner/miniforge3/conda-bld/lhapdf_1724108231610/test_tmp/examples/analyticpdf'
  Reason: tried: '/usr/local/lib/libLHAPDF.dylib' (no such file), '/usr/lib/libLHAPDF.dylib' (no such file)
```

so `rpath` is getting messed up somewhere.

Comment (matthewfeickert) [reply recipe/run_test.sh]: I don't actually understand how `libtool` works, but I think given some `git grep`ing in https://gitlab.com/hepcedar/lhapdf that this might(?) be related to `sys_lib_search_path_spec` in

https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/m4/libtool.m4#L2653-2666

That doesn't make sense though as

```bash
./configure --prefix=$PREFIX
```

should have set everything correctly.

File: recipe/build.sh
Comment (chrisburr) [reply recipe/build.sh]: Not fully, maybe it should be `-I` juding by: https://cython.readthedocs.io/en/latest/src/userguide/sharing_declarations.html#search-paths-for-definition-files

My guess is a change in the import behavour in cython 3 but it's not totally clear from their docs. Looking at https://github.com/cython/cython/issues/3862 it seems like it's intentional.

File: recipe/run_test.sh
Comment (chrisburr) [reply recipe/run_test.sh]: My first guess would be you need to use `$CXXFLAGS` as well.

File: recipe/build.sh
Comment (matthewfeickert) [reply recipe/build.sh]: > Not fully, maybe it should be `-I` juding by: https://cython.readthedocs.io/en/latest/src/userguide/sharing_declarations.html#search-paths-for-definition-files

Ah nice. I'll get that in my rebase.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/run_test.sh
Comment (matthewfeickert) [inline recipe/run_test.sh]:
```suggestion
./hessian2replicas MSTW2008nnlo68cl 1234 10
```

Make this just `10` to avoid wasting time here.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh
Comment (matthewfeickert) [inline recipe/build.sh:22]: Adding `CXXFLAGS` here has an effect on `lhapdf-config --cxxflags` but in testing on my M2 machine adding `LDFLAGS="${LDFLAGS}"` to the `configure` command as well didn't have any affect on `lhapdf-config --ldflags` so I gave up and went with adding `-Wl,-rpath,"$(lhapdf-config --libdir)"` to all compile steps by hand. (c.f. examples in `run_test.sh`).

File: recipe/meta.yaml
Comment (matthewfeickert) [inline recipe/meta.yaml:44]: NumPy is added here as it is needed by `pythonexample.py` (c.f. https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/examples/pythonexample.py#L26).

File: recipe/run_test.sh
Comment (matthewfeickert) [inline recipe/run_test.sh:70]: Using the `-Wl,-rpath,"$(lhapdf-config --libdir)"` pattern here as I couldn't figure out how to set `-Wl,-rpath,$PREFIX/lib` at configure time in a way that would add this information to `lhapdf-config --ldflags`.

File: recipe/meta.yaml
Comment (scarlehoff) [inline recipe/meta.yaml]: Is it possible to remove the `>=3.0.0` limitation from here?

There (might) be other projects that require cython < 3.

File: recipe/meta.yaml
Comment (scarlehoff) [inline recipe/meta.yaml:34]: Why is it necessary to remove the `# [build_platform != target_platform]`? These were not needed in Linux before I think.

File: recipe/build.sh
Comment (scarlehoff) [inline recipe/build.sh:20]:
```suggestion
```

**Comment by matthewfeickert** [comment]
@conda-forge/lhapdf this is ready for review now. Let me know if you have any questions, though hopefully the tests I've added should make this easier to review.

The only files that are worth looking at here are:
* `recipe/meta.yaml`
* `recipe/build.sh`
* `recipe/run_test.sh`

everything else is just the result of rerenders.

@chrisburr Any additional review or feedback you can offer is very welcome.

File: recipe/build.sh
Comment (scarlehoff) [inline recipe/build.sh]: Are these two needed in non-arm apple? (because they were not necessary before)

**Review by scarlehoff — COMMENTED** [review]
Added a few questions, I'd like to reduce as much as possible any extra dependencies if they are not absolutely needed.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml]: These are `host` requirements, not `run` requirements. `build` and `host` only affect this build process.

c.f. https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#host

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml:34]: @chrisburr removed these, but the pattern wasn't necessary. `# [build_platform != target_platform]` is only useful when you're cross compiling and need the `.so`s. These are all build tools where that won't ever happen.

c.f. https://conda-forge.org/docs/maintainer/knowledge_base/#details-about-cross-compiled-python-packages

File: recipe/build.sh
Comment (matthewfeickert) [reply recipe/build.sh:20]: Having `./configure --help` in the logs is useful for debugging and doesn't hurt.

File: recipe/meta.yaml
Comment (scarlehoff) [reply recipe/meta.yaml:34]: But as far as I am aware these are not used or needed in the linux (or mac for that matter) builds right?

File: recipe/meta.yaml
Comment (scarlehoff) [reply recipe/meta.yaml]: Are you sure about the host? I confess it has never been clear to me how much the host requirements leaks into the final package.

Still, if the requirement is not real (and it was working fine with cython < 3) I don't see the point of enforcing it.

File: recipe/build.sh
Comment (matthewfeickert) [reply recipe/build.sh]: I can't find a good diff to compare with, so let me push a commit that breaks this out into a `arm64` block again and see.

Comment (scarlehoff) [reply recipe/build.sh]: Okay, thanks!

I believe the closer this is to the "default" installation in as many systems as possible, the easier it will be to accommodate future changes.

File: recipe/meta.yaml
Comment (chrisburr) [reply recipe/meta.yaml:34]: I needed it for a native macOS build due to: https://github.com/conda-forge/lhapdf-feedstock/blob/d89b118c527edee53db55b2fbb372be52411c9dd/recipe/build.sh#L11

Potentially the requirement should be the same in the `build.sh`. Though I'd also suggest always cythonising the sources as there isn't much of a downside and it could be useful in cases like new python versions.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml
Comment (matthewfeickert) [inline recipe/meta.yaml:10]: Building from GitLab means that you're starting from the source with no additional changes like the `configure` script being already generated (as the hepforge releases do), so you get a more uniform experience of building across all platforms. This is also going to require that all platforms install the autotools requirements as `build` requirements.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml:34]: This PR now changes to using the GitLab source for build uniformity, which requires these now, and so gives the same experience that @chrisburr had. It is also important to point out that in the future there could and probably will be native Apple silicon runners on Azure pipelines, and so this means that future builds will work on those native architecture runners.

Comment (scarlehoff) [reply recipe/meta.yaml:34]: You mean you needed it when running the conda-recipe locally?

I'd really prefer to keep the build/installation as close to the "official installation" as possible.

File: recipe/build.sh
Comment (matthewfeickert) [inline recipe/build.sh:17]: Building from GitLab means that you're starting from the source with no additional changes like the `configure` script being already generated (as the hepforge releases do), so you get a more uniform experience of building across all platforms. This means that you need to generate the `configure` script though.

File: recipe/build.sh
Comment (matthewfeickert) [inline recipe/build.sh:18]: Generate the Python sources with Cython for all build platforms given @chrisburr's recommendation that this can be useful in cases like new Python versions.

Comment (scarlehoff) [inline recipe/build.sh:18]: As said elsewhere, I'd really prefer not touch the original files if it is not necessary (which as far as I understand it's only the case for apple) so that what the user gets is as close as possible to what they'd get installing directly from source. Bugs included.

**Review by matthewfeickert — COMMENTED** [review]
Switching the build source to GitLab gives a more uniform build experience across all platforms, as hepforge distributions had the `configure` script pre-built (not part of the GitLab source) but it isn't compatible across all platforms. Using the GitLab source also makes it very clear that this is an `autotools` build project and that the build requirements need to be provided for it (also now uniform across platform).

This now also places only a lower bound of `cython >=0.24` as that's what `configure` explicitly checks for. Given @chrisburr's recommendation all platforms now use Cython to build the Python source files.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml]: > Are you sure about the host? 

Yes.

**Comment by scarlehoff** [comment]
The problem for me is that if the official releases of LHAPDF are those from hepforge, I wouldn't want to move from those for the sake of uniformity across platform.

To be more specific, I think what someone installing this package expects is that it is the same package that they get when downloading from https://lhapdf.hepforge.org.
How the package is built or whether the dependencies are the same across platforms is only important for us nerds...

**Comment by scarlehoff** [comment]
Or, let me rephrase the question/comment. Why are the sources from https://lhapdf.hepforge.org different and which ones should be taken as the real "LHAPDF release".

File: recipe/meta.yaml
Comment (scarlehoff) [reply recipe/meta.yaml:34]: > This PR now changes to using the GitLab source for build uniformity, which requires these now,

Ah, gotcha. I missed that when writing the comment above. That resolves the question.

I need to think about moving from hepforge to gitlab because I'm not convinced, but that clarifies most of my questions / points.

Comment (matthewfeickert) [reply recipe/meta.yaml:34]: > You mean you needed it when running the conda-recipe locally?

No, @chrisburr means building on an aarch64 macOS machine. If you get the source (GitLab or hepforce) you are required to generate a configure script using `autoreconf` (or `autoconf`) as the GitLab source doesn't provide a `configure` script and the hepforge provided `configure` script doesn't know how to support `aarch64` macOS. In both situations a native build on aarch64 macOS requires doing the "normal" autotools projects procedure of generating `configure`.

> I'd really prefer to keep the build/installation as close to the "official installation" as possible.

This is just providing the tools to build the source code. How is this different than the official installation procedure?

Comment (scarlehoff) [reply recipe/meta.yaml:34]: > No, @chrisburr means building on an aarch64 macOS machine.

Which must've beeen local (or at least, not the CI)!

In any case, I was missing the gitlab part (which either wasn't here yesterday or I missed it) to see it was needed now also in linux.

**Comment by matthewfeickert** [comment]
> To be more specific, I think what someone installing this package expects is that it is the same package that they get when downloading from https://lhapdf.hepforge.org/.

They are, AFAIK. Installing `lhapdf` from conda-forge gets you only the built C++ library, Python bindings, and the `run` requirements. Using autotools to generate the configure scripts for an autotools project shouldn't change anything.

> Or, let me rephrase the question/comment. Why are the sources from https://lhapdf.hepforge.org different and which ones should be taken as the real "LHAPDF release".

Ah good question. As to _why_ we'll have to ask @agbuckley on that. My guess(?) is that Andy is trying to be nice here to people and not have them do additional `autotools` shenanigans by prebuilding the `configure` script they would have to generate themselves. Though I'll have to defer to him as to why the hepforge and GitLab sources aren't 1-to-1.

**Comment by scarlehoff** [comment]
If the hepforge sources are supposed to be _just_ the gitlab release with a configure, or if the gitlab release are the official "release" going forward (6.5.4 is the only one with a release there https://gitlab.com/hepcedar/lhapdf/-/releases), I'd be ok installing from there.

But we have quite a few pieces of software that depend on LHAPDF so I'd like to be sure before.

> Using autotools to generate the configure scripts for an autotools project shouldn't change anything.

I don't know if what's in the hepforge tarball is _just_ the hepforge release + autotools or whether some patches are applied, some changes backported, etc.

File: recipe/meta.yaml
Comment (matthewfeickert) [reply recipe/meta.yaml:34]: > Which must've been local (or at least, not the CI)!

Ah yes, correct. Sorry I didn't make that clear.

> I was missing the gitlab part (which either wasn't here yesterday or I missed it) to see it was needed now also in linux.

You're correct that was a today change. I apologize about the rebasing as that can make some of the tracking of changes harder to follow across time, but easier to look at if you're looking at each commit in a review. I'll update the PR body and then also leave comments in a "review" about how and why the changes were made, but to make things less noisy I'll not do any further rebasing until there's a consensus on what direction to move things towards. :+1:

Comment (scarlehoff) [reply recipe/meta.yaml:34]: No worries, I should've waited until it was finished to look through the changes!

**Comment by matthewfeickert** [comment]
> I don't know if what's in the hepforge tarball is just the hepforge release + autotools or whether some patches are applied, some changes backported, etc.

Very good point that I embarrassingly did not think about! I'll again defer to @agbuckley and @20DM on this, but the following `diff_sources.sh` check:

```shell
#!/bin/bash

if [ -d lhapdf-6.5.4_gitlab ]; then
    rm -rf lhapdf-6.5.4_gitlab
fi
curl -sL0 "https://gitlab.com/hepcedar/lhapdf/-/archive/lhapdf-6.5.4/lhapdf-lhapdf-6.5.4.tar.gz" | tar -xz
mv lhapdf-lhapdf-6.5.4 lhapdf-6.5.4_gitlab

if [ -d lhapdf-6.5.4_hepforge ]; then
    rm -rf lhapdf-6.5.4_hepforge
fi
curl -sL "https://lhapdf.hepforge.org/downloads/?f=LHAPDF-6.5.4.tar.gz" -o LHAPDF-6.5.4.tar.gz
tar -xzf LHAPDF-6.5.4.tar.gz
mv LHAPDF-6.5.4 lhapdf-6.5.4_hepforge

diff -bur lhapdf-6.5.4_gitlab/ lhapdf-6.5.4_hepforge/ &> lhapdf-6.5.4_diff.txt

grep "_gitlab" lhapdf-6.5.4_diff.txt &> only_in_gitlab.txt
grep "_hepforge" lhapdf-6.5.4_diff.txt &> only_in_hepforge.txt
```

produces the following

[lhapdf-6.5.4_diff.txt](https://github.com/user-attachments/files/16681017/lhapdf-6.5.4_diff.txt)

<details>
<summary>lhapdf-6.5.4_diff.txt:</summary>

```
Only in lhapdf-6.5.4_hepforge/: aclocal.m4
Only in lhapdf-6.5.4_hepforge/bin: Makefile.in
Only in lhapdf-6.5.4_hepforge/: config
Only in lhapdf-6.5.4_hepforge/: configure
Only in lhapdf-6.5.4_gitlab/: countnsets
Only in lhapdf-6.5.4_gitlab/doc: DoxyLayout.xml
Only in lhapdf-6.5.4_gitlab/doc: examples.dox
Only in lhapdf-6.5.4_gitlab/doc: lhapdf.css
Only in lhapdf-6.5.4_gitlab/doc: main.dox
Only in lhapdf-6.5.4_hepforge/doc: Makefile.in
Only in lhapdf-6.5.4_hepforge/examples: analyticpdf
Only in lhapdf-6.5.4_hepforge/examples: compatibility
Only in lhapdf-6.5.4_hepforge/examples: hessian2replicas
Only in lhapdf-6.5.4_hepforge/examples: Makefile.in
Only in lhapdf-6.5.4_hepforge/examples: reweight
Only in lhapdf-6.5.4_hepforge/examples: testpdf
Only in lhapdf-6.5.4_hepforge/examples: testpdfset
Only in lhapdf-6.5.4_hepforge/examples: testpdfunc
Only in lhapdf-6.5.4_gitlab/: .gitignore
Only in lhapdf-6.5.4_gitlab/: .gitlab-ci.yml
Only in lhapdf-6.5.4_hepforge/include/LHAPDF: Makefile.in
Only in lhapdf-6.5.4_hepforge/include/LHAPDF: Version.h
Only in lhapdf-6.5.4_hepforge/include: Makefile.in
Only in lhapdf-6.5.4_hepforge/: lhapdf.pc
Only in lhapdf-6.5.4_gitlab/: LICENSE
Only in lhapdf-6.5.4_gitlab/m4: binreloc.m4
Only in lhapdf-6.5.4_hepforge/m4: lt~obsolete.m4
Only in lhapdf-6.5.4_hepforge/m4: ltoptions.m4
Only in lhapdf-6.5.4_hepforge/m4: ltsugar.m4
Only in lhapdf-6.5.4_hepforge/m4: ltversion.m4
Only in lhapdf-6.5.4_gitlab/m4: pkgchecks.m4
Only in lhapdf-6.5.4_gitlab/m4: swig.m4
Only in lhapdf-6.5.4_hepforge/: Makefile.in
Only in lhapdf-6.5.4_gitlab/: migration
Only in lhapdf-6.5.4_gitlab/: plotevratios.py
Only in lhapdf-6.5.4_gitlab/: plotnsets
Only in lhapdf-6.5.4_hepforge/src: Makefile.in
Only in lhapdf-6.5.4_hepforge/src/yamlcpp: Makefile.in
Only in lhapdf-6.5.4_gitlab/: testinfo
Only in lhapdf-6.5.4_hepforge/tests: Makefile.in
Only in lhapdf-6.5.4_gitlab/tests: testgridpdfinit.cc
Only in lhapdf-6.5.4_hepforge/wrappers: Makefile.in
Only in lhapdf-6.5.4_hepforge/wrappers/python: lhapdf.cpp
Only in lhapdf-6.5.4_hepforge/wrappers/python: Makefile.in
```

</details>

[only_in_gitlab.txt](https://github.com/user-attachments/files/16681018/only_in_gitlab.txt)


<details>
<summary>only_in_gitlab.txt:</summary>

```
Only in lhapdf-6.5.4_gitlab/: countnsets
Only in lhapdf-6.5.4_gitlab/doc: DoxyLayout.xml
Only in lhapdf-6.5.4_gitlab/doc: examples.dox
Only in lhapdf-6.5.4_gitlab/doc: lhapdf.css
Only in lhapdf-6.5.4_gitlab/doc: main.dox
Only in lhapdf-6.5.4_gitlab/: .gitignore
Only in lhapdf-6.5.4_gitlab/: .gitlab-ci.yml
Only in lhapdf-6.5.4_gitlab/: LICENSE
Only in lhapdf-6.5.4_gitlab/m4: binreloc.m4
Only in lhapdf-6.5.4_gitlab/m4: pkgchecks.m4
Only in lhapdf-6.5.4_gitlab/m4: swig.m4
Only in lhapdf-6.5.4_gitlab/: migration
Only in lhapdf-6.5.4_gitlab/: plotevratios.py
Only in lhapdf-6.5.4_gitlab/: plotnsets
Only in lhapdf-6.5.4_gitlab/: testinfo
Only in lhapdf-6.5.4_gitlab/tests: testgridpdfinit.cc
```

</details>

[only_in_hepforge.txt](https://github.com/user-attachments/files/16681019/only_in_hepforge.txt)

<details>
<summary>only_in_hepforge.txt:</summary>

```
Only in lhapdf-6.5.4_hepforge/: aclocal.m4
Only in lhapdf-6.5.4_hepforge/bin: Makefile.in
Only in lhapdf-6.5.4_hepforge/: config
Only in lhapdf-6.5.4_hepforge/: configure
Only in lhapdf-6.5.4_hepforge/doc: Makefile.in
Only in lhapdf-6.5.4_hepforge/examples: analyticpdf
Only in lhapdf-6.5.4_hepforge/examples: compatibility
Only in lhapdf-6.5.4_hepforge/examples: hessian2replicas
Only in lhapdf-6.5.4_hepforge/examples: Makefile.in
Only in lhapdf-6.5.4_hepforge/examples: reweight
Only in lhapdf-6.5.4_hepforge/examples: testpdf
Only in lhapdf-6.5.4_hepforge/examples: testpdfset
Only in lhapdf-6.5.4_hepforge/examples: testpdfunc
Only in lhapdf-6.5.4_hepforge/include/LHAPDF: Makefile.in
Only in lhapdf-6.5.4_hepforge/include/LHAPDF: Version.h
Only in lhapdf-6.5.4_hepforge/include: Makefile.in
Only in lhapdf-6.5.4_hepforge/: lhapdf.pc
Only in lhapdf-6.5.4_hepforge/m4: lt~obsolete.m4
Only in lhapdf-6.5.4_hepforge/m4: ltoptions.m4
Only in lhapdf-6.5.4_hepforge/m4: ltsugar.m4
Only in lhapdf-6.5.4_hepforge/m4: ltversion.m4
Only in lhapdf-6.5.4_hepforge/: Makefile.in
Only in lhapdf-6.5.4_hepforge/src: Makefile.in
Only in lhapdf-6.5.4_hepforge/src/yamlcpp: Makefile.in
Only in lhapdf-6.5.4_hepforge/tests: Makefile.in
Only in lhapdf-6.5.4_hepforge/wrappers: Makefile.in
Only in lhapdf-6.5.4_hepforge/wrappers/python: lhapdf.cpp
Only in lhapdf-6.5.4_hepforge/wrappers/python: Makefile.in
```

</details>

So it looks like in the hepforge distribution the autotools generation steps have already happened and examples have been built. The only source file that is different (ignoring the existence of `Makefile.in`s as those are build products) is `include/LHAPDF/Version.h` existing in hepforge, but looking at that file we see it is generated from `configure` 

```console
$ cat lhapdf-6.5.4_hepforge/include/LHAPDF/Version.h
/* include/LHAPDF/Version.h.  Generated from Version.h.in by configure.  */
// -*- C++ -*-
//
// This file is part of LHAPDF
// Copyright (C) 2012-2016 The LHAPDF collaboration (see AUTHORS for details)
//
#pragma once
#ifndef LHAPDF_Version_H
#define LHAPDF_Version_H

#include <string>

/* "LHAPDF version string" */
#define LHAPDF_VERSION "6.5.4"

/* "LHAPDF version as an int" */
#define LHAPDF_VERSION_CODE 60504

/* "Whether PDFLIB Fortran compatibility is available" */
#define LHAPDF_PDFLIB 1

/* "Whether LHAPDF5 C++ compatibility is available" */
#define LHAPDF_LHA5CXX 1

// Separate int-valued macro for conditional compilation. Doesn't exist in LHAPDF5.
// Use like "#if defined LHAPDF_MAJOR_VERSION && LHAPDF_MAJOR_VERSION == 6 ..."
#define LHAPDF_MAJOR_VERSION 6

namespace LHAPDF {


  /// Get the LHAPDF library version code (as a string)
  inline std::string version() {
    return LHAPDF_VERSION;
  }


}
#endif
```

and if I diff a local build vs. it

```
diff build_artifacts/pkg_cache/lhapdf-6.5.4-py312h2e396b9_5/include/LHAPDF/Version.h /tmp/diff-sources/lhapdf-6.5.4_hepforge/include/LHAPDF/Version.h  # no difference
```

there's no difference.

**Comment by agbuckley** [comment]
The whole design of autotools is that it builds tarballs that _don't_ require any of the autotools to be installed, i.e. those are only for templating etc. purposes and the resulting archive can be unpacked and run with `sh` and `make` (plus whatever tools the `configure` shell script tests for).

**Comment by scarlehoff** [comment]
hi @agbuckley, then you confirm the releases from hepforge are just the "`make dist`" of gitlab?

**Comment by matthewfeickert** [comment]
> @agbuckley, then you confirm the releases from hepforge are just the "make dist" of gitlab?

@agbukcley, small ping here for your thoughts on this.

**Comment by matthewfeickert** [comment]
> @agbuckley, then you confirm the releases from hepforge are just the "make dist" of gitlab?

@agbuckley, small ping here for your thoughts on this. (Sorry, this is a repost where I don't typo this time.)

**Comment by agbuckley** [comment]
> > @agbuckley, then you confirm the releases from hepforge are just the "make dist" of gitlab?
> 
> @agbuckley, small ping here for your thoughts on this. (Sorry, this is a repost where I don't typo this time.)

Yes that's right. (Sorry, 24h travel yesterday...)

**Comment by matthewfeickert** [comment]
> > ... then you confirm the releases from hepforge are just the "make dist" of gitlab?
> 
> Yes that's right.

@conda-forge/lhapdf given this, what are your thoughts on things should move forward?

**Comment by scarlehoff** [comment]
Funnily enough I was also in a >24h trip :P. I'll have a look & merge next week.

**Review by scarlehoff — APPROVED** [review]
Thanks for this! lgtm!

ps: just a heads up @radonirinaunimi that this changes the installation of LHAPDF from the site's package to the gitlab. Should not make a difference for pineappl but just so you know.

**Pull request merged**
